### PR TITLE
Revive the Elastic Beanstalk dev server

### LINKED ISSUE #462: Bring back the auto-deploying dev server
I miss the old dev server that got bumped on each commit. (Which was killed in aptly named #404 PR):

> _"Going to retire idea-town-dev, because we're moving to a deployment model supported by ops."_

We have a stage server, https://testpilot.stage.mozaws.net/, but it currently has an old (yesterday's) build and won't let me download the signed addon (it points to a 404 page on GitHub -- see https://github.com/mozilla/testpilot/issues/244). Currently I'm not 100% certain how I can build the addon myself so I can log into stage TestPilot and check the UI.

It'd be nice if we had a:
1. An auto-deployed dev server which is updated on each commit and is always fresh, so QA can kick the tires and test stuff out.
2. A stage environment which is safe and can be promoted up to prod.
3. A prod environment to share our gift with the world.

I think we have a solid plan for 2) and 3), so I guess I'm just lobbying to get the self aware, auto-deploying dev server back so I don't need to throw up an entire Docker stack just to look at the UI.

---

@mostlygeek made the following, excellent statements earlier in IRC:

> 9:53 AM **mostlygeek:** pdehaan: did you want me to deploy the latest master to stage? 
> 9:53 AM **mostlygeek:** will that eventually make it to prod? 
> 9:53 AM **mostlygeek:** or is this just a thing to play around with? 
> 9:53 AM **pdehaan**: i just want to play around.
> 9:54 AM **pdehaan**: and yes please (re: deploy). it looks like the last 2 PRs are red, so we won't wait around for them.
> 9:54 AM **mostlygeek**: pdehaan: the stage deployment isn't really the right thing to push something to play around with
> 9:54 AM **mostlygeek**: we need a dev account for that
> 9:54 AM **mostlygeek**: or a dev thing.. 
> 9:55 AM **mostlygeek**: probably something where master is auto-pushed to it 
> 9:55 AM **pdehaan**: we used to have a dev server, but it got killed.
> 9:55 AM **pdehaan**: ok, maybe leave it for now, and we can figure out a plan on how QA can test this.

### PR DESCRIPTION
**Pull request by lmorchard**
- Deploy to Elastic Beanstalk from Docker Hub via Circle CI
- Add dev to addon environments

Fixes #462

### CODE REVIEW

File: .ebextensions/00-environment.config-dist [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  - option_name: INITIAL_ADMIN_EMAIL
+    value: lorchard@mozilla.com
+
+  # SECRET_KEY *must* have a unique, secret value
+  - option_name: SECRET_KEY
+    value: YOUR SECRET KEY HERE
+
+  # Expected hostname(s) for the site
+  - option_name: ALLOWED_HOSTS
+    value: ideatown.firefox.com
```
Comment (pdehaan) [inline .ebextensions/00-environment.config-dist]: what dis?

File: .ebextensions/00-environment.config-dist [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+  # URL to the current version of the Idea Town addon
+  - option_name: ADDON_URL
+    value: http://example.com/idea-town.xpi
```
Comment (pdehaan) [inline .ebextensions/00-environment.config-dist]: and dis?

**Comment by pdehaan** [comment]
Do we need to relax the [`ACCOUNT_AUTOACTIVATION_DOMAINS`](https://github.com/mozilla/testpilot/pull/449/files#diff-461aaf29fb21fa9e18300c35ee7853ccR153) thing to allow @restmail accounts too?

**Comment by lmorchard** [comment]
Seems like this stuff successfully deployed via my personal Circle CI account over here:

http://testpilot-dev.us-east-1.elasticbeanstalk.com/

File: .ebextensions/00-environment.config-dist (hunk 2 above)
Comment (lmorchard) [inline .ebextensions/00-environment.config-dist]: This is just the example file. The real settings are totally filled in differently than this and uploaded to a private S3 bucket

**Comment by lmorchard** [comment]
> Do we need to relax the ACCOUNT_AUTOACTIVATION_DOMAINS thing to allow @restmail accounts too?

Nope, that should be turned off in the dev env vars on elastic beanstalk. Can't really show the settings here, though, because they're littered with private credentials

**Comment by lmorchard** [comment]
I'll change a few of the example settings so they're less confusing, though

**Comment by pdehaan** [comment]
This is awesome, thanks!

**Comment by pdehaan** [comment]
@lmorchard OK, i think I'm seeing one small issue.

I can go to http://testpilot-dev.us-east-1.elasticbeanstalk.com/ and it will prompt me to log in (which I do with my @mozilla) and then install the add-on.
I then install said add-on but it never lets me proceed (despite the add-on being installed -- about:addons reports I have "Test Pilot 0.4.10-dev-69109cb" installed).

Also, the "Environment" dropdown only shows "Local", "Staging", and "Production" and not the new "Dev" target you added in [addon/package.json](https://github.com/mozilla/testpilot/pull/524/files#diff-18ea69015717516d88dedc84076c033eR51).

**Comment by lmorchard** [comment]
Yeah, that add-on thing is because the auto-deployment comes from whatever was the latest Docker Hub build from master branch. That doesn't include this PR.

So, the latest on master is 69109cb - as included in that add-on version string. But, this PR is 18af7ee153f7b6e87d75029b4298ba824e96c2db and master doesn't have the changes from this PR yet.

**Comment by lmorchard** [comment]
Also, FWIW, you _should_ be able to login without an mozilla.com address on this dev server.

**Comment by pdehaan** [comment]
And will this server auto-deploy on each commit? Or do you kick it daily/manually?
I vote to merge this and then I can retest everything once DockerHub rebuilds master and the "Dev" env config is in the add-on.

**Comment by lmorchard** [comment]
> And will this server auto-deploy on each commit?

Yup, should be automatic! Once this PR is merged, the [line in `circle.yml` with `deploy-dev.sh`](https://github.com/mozilla/testpilot/pull/524/files#diff-29944324a3cbf9f4bd0162dfe3975d88R87) should kick in for every subsequent merge to master and update this dev server with the latest on master.

**Comment by lmorchard** [comment]
Hmm, this shouldn't break anything but itself, so I think I will just go ahead and hit the green button myself!

**Comment by pdehaan** [comment]
Related to some stuff above, but I'm happy to file a separate bug if that is cleaner. I was able to successfully install the addon and configure it for the Dev stack, but now I'm seeing 3x 404s (presumably one for each experiment).

| &nbsp; | URL | RESPONSE |
| :-: | --- | --- |
| GET | https://idea-town-media-prod.s3.amazonaws.com/demo/experiments_experiment/7/f/7fb027cc91d81bfe14d0c7daf2ada413_thumbnail_1449676075_0620.png | [HTTP/1.1 404 Not Found 258ms] |
| GET | https://idea-town-media-prod.s3.amazonaws.com/demo/experiments_experiment/a/8/a8182b008f6434ba702ab177232625e2_thumbnail_1449675941_0564.png | [HTTP/1.1 404 Not Found 263ms] |
| GET | https://idea-town-media-prod.s3.amazonaws.com/demo/experiments_experiment/e/1/e19739ea25188c866b654d3aa2616b27_thumbnail_1449778314_0632.png | [HTTP/1.1 404 Not Found 258ms] |

**Pull request merged**